Magento/Tests/Classes/ConstructorProxyInterceptorUnitTest.2.inc

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Vendor\Module\Proxy;
+
+use \Vendor\Module\Proxy\AnyName as ProxAlias;
+use \Exception as SafeAlias;
+
+class AnyName {}
+
+/**
+ * // Rule find: constructor use of proxy class
+ */
+class Foo
```

Comment:
Proxy classes are the one which have `\Proxy` suffix. Interceptor classes always have `\Interceptor` suffix. These can be accurately detected and should raise error.


Plugin classes have arbitrary names and thus are difficult to catch. These rules may have accuracy issues and can raise warnings.

Reply:
As this is based on names of constructor types and variables there will be a false-positive issue. In discussion with @lenaorobei it was felt that keeping these as warnings was of utility despite this.

Reply:
Even is it's bad naming it's worth pointing developers that class name with `Interceptor` and `Proxy` suffix is not a good idea.